src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToTestAMultitargetedSolutionWithPublishReleaseOrPackRelease.cs

```diff
@@ -0,0 +1,395 @@
+            var sln = solutionAndProjects.Item1;
+
+            new DotnetPublishCommand(Log)
+                .WithWorkingDirectory(Directory.GetParent(sln.SolutionPath).FullName) // code under test looks in CWD, ensure coverage outside this scenario
+                .Execute(sln.SolutionPath)
+                .Should()
+                .Fail()
+                .And
+                .HaveStdErrContaining(string.Format(Strings.SolutionProjectConfigurationsConflict, PublishRelease, ""));
+        }
+
+        [Fact]
+        public void It_doesnt_error_if_environment_variable_opt_out_enabled_but_PublishRelease_conflicts()
+        {
+            var tfm = ToolsetInfo.CurrentTargetFramework;
+            var solutionAndProjects = Setup(Log, new List<string> { tfm }, new List<string> { tfm }, PublishRelease, "true", "false");
+            var sln = solutionAndProjects.Item1;
+
+            new DotnetPublishCommand(Log)
+                .WithEnvironmentVariable("DOTNET_CLI_DISABLE_PUBLISH_AND_PACK_RELEASE", "true")
+                .Execute(sln.SolutionPath) // This property won't be set in VS, make sure the error doesn't occur because of this by mimicking behavior.
+                .Should()
+                .Pass();
+        }
+
+        [Fact]
+        public void It_packs_with_Release_on_all_TargetFrameworks_If_8_or_above_is_included()
+        {
+            var testProject = new TestProject()
+            {
+                IsExe = true,
+                TargetFrameworks = "net7.0;net8.0"
+            };
+            testProject.RecordProperties("Configuration");
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            new DotnetPackCommand(Log)
+                .WithWorkingDirectory(Path.Combine(testAsset.TestRoot, testProject.Name))
+                .Execute()
+                .Should()
+                .Pass();
+
+            var properties = testProject.GetPropertyValues(testAsset.TestRoot, targetFramework: "net7.0", configuration: "Release"); // this will fail if configuration is debug and TFM code didn't work.
+            string finalConfiguration = properties["Configuration"];
+            finalConfiguration.Should().BeEquivalentTo("Release");
+        }
+
+        private void VerifyCorrectConfiguration(List<Dictionary<string, string>> finalProperties, string expectedConfiguration)
+        {
+            string expectedBooleanValue = "true";
+            if(expectedConfiguration != "Release")
+            {
+                expectedBooleanValue = "false";
+            }
+
+
+            Assert.Equal(expectedBooleanValue, finalProperties.First()[Optimize]);
+            Assert.Equal(expectedConfiguration, finalProperties.First()[Configuration]);
+
+            Assert.Equal(expectedBooleanValue, finalProperties.ElementAt(1)[Optimize]);
+            Assert.Equal(expectedConfiguration, finalProperties.ElementAt(1)[Configuration]);
```

Comment:
I'd use `finalProperties[0]` and `finalProperties[1]` instead of `.First()` and `ElementAt(1)`.  The methods work on enumerables, but for lists and arrays you can use direct indexing.